ya_glm/opt/constraint/convex.py

```diff
@@ -58,8 +75,9 @@ def is_proximable(self):
 # See https://gist.github.com/mblondel/6f3b7aaad90606b98f71
 # for more algorithms.
 def project_simplex(v, z=1):
-    if np.sum(v) <= z:
-        return v
+    # z is what the entries need to add up to, e.g. z=1 for probability simplex
+    if np.sum(v) <= z:  # don't we want the simplex to mean sum == z not sum <= z?
```

Comment:
Agreed, but perhaps up to machine precision.

Reply:
So we can just delete this sum <= z check?

Reply:
Yes on second thought anytime we are calling project it's probably safe to assume we were not originally on the simplex.

Reply:
Fixed